monoio-http/src/h1/codec/server.rs

```diff
@@ -14,12 +15,12 @@ pub struct ServerCodec<IO: AsyncWriteRent> {
 }
 
 impl<IO: Split + AsyncReadRent + AsyncWriteRent> ServerCodec<IO> {
-    pub fn new(io: IO) -> Self {
+    pub fn new(io: IO, timeout: Duration) -> Self {
```

Comment:
Timeout should be a optional config, and should be zero cost if not used.

Reply:
Not sure how to make it really zero cost, but I added if else around the timeout option so that we can avoid doing timeout when it's None.